spark/src/main/scala/geotrellis/spark/viewshed/IterativeViewshed.scala

```diff
@@ -0,0 +1,390 @@
+/*
+ * Copyright 2017 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.spark.viewshed
+
+import geotrellis.proj4.LatLng
+import geotrellis.raster._
+import geotrellis.raster.rasterize.Rasterizer
+import geotrellis.raster.viewshed.R2Viewshed
+import geotrellis.raster.viewshed.R2Viewshed._
+import geotrellis.spark._
+import geotrellis.spark.tiling._
+import geotrellis.util._
+import geotrellis.vector._
+
+import com.vividsolutions.jts.{ geom => jts }
+import org.apache.log4j.Logger
+import org.apache.spark.rdd.RDD
+import org.apache.spark.SparkContext
+import org.apache.spark.storage.StorageLevel
+import org.apache.spark.util.AccumulatorV2
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+
+/**
+  * A Spark-enabled implementation of R2 [1] viewshed.
+  *
+  * 1. Franklin, Wm Randolph, and Clark Ray.
+  *    "Higher isn’t necessarily better: Visibility algorithms and experiments."
+  *    Advances in GIS research: sixth international symposium on spatial data handling. Vol. 2.
+  *    Taylor & Francis Edinburgh, 1994.
+  */
+object IterativeViewshed {
+
+  /**
+    * A `Point6D` is an array of six doubles.  The indices are mapped as follows:
+    *   0: x-coordinate (in the units used by the layer)
+    *   1: y-coordinate (in the units used by the layer)
+    *   2: view height (in units of "meters")
```

Comment:
Looking at the algorithm this is height from the surface at the given point. Is that what we expect the definition to be or is it expected to be absolute height? + @lossyrob @rossbernet